mdp.ipynb

```diff
@@ -682,40 +682,40 @@
+    "It takes as inputs two parameters, an MDP to solve and epsilon, the maximum error allowed in the utility of any state. It returns a dictionary containing utilities where the keys are the states and values represent utilities. <br> Value Iteration starts with arbitrary initial values for the utilities, calculates the right side of the Bellman equation and plugs it into the left hand side, thereby updating the utility of each state from the utilities of its neighbors. \n",
     "This is repeated until equilibrium is reached. \n",
-    "It works on the principle of _Dynamic Programming_. \n",
-    "If U_i(s) is the utility value for state _s_ at the _i_ th iteration, the iteration step, called Bellman update, looks like this:\n",
+    "It works on the principle of _Dynamic Programming_ - using precomputed information to simplify the subsequent computation. \n",
+    "If $U_i(s)$ is the utility value for state $s$ at the $i$ th iteration, the iteration step, called Bellman update, looks like this:\n",
     "\n",
     "$$ U_{i+1}(s) \\leftarrow R(s) + \\gamma \\max_{a \\epsilon A(s)} \\sum_{s'} P(s'\\ |\\ s,a)U_{i}(s') $$\n",
     "\n",
     "As you might have noticed, `value_iteration` has an infinite loop. How do we decide when to stop iterating? \n",
     "The concept of _contraction_ successfully explains the convergence of value iteration. \n",
     "Refer to **Section 17.2.3** of the book for a detailed explanation. \n",
-    "In the algorithm, we calculate a value _delta_ that measures the difference in the utilities of the current time step and the previous time step. \n",
+    "In the algorithm, we calculate a value $delta$ that measures the difference in the utilities of the current time step and the previous time step. \n",
     "\n",
     "$$\\delta = \\max{(\\delta, \\begin{vmatrix}U_{i + 1}(s) - U_i(s)\\end{vmatrix})}$$\n",
     "\n",
-    "This value of delta decreases over time.\n",
-    "We terminate the algorithm if the delta value is less than a threshold value determined by the hyperparameter _epsilon_.\n",
+    "This value of delta decreases as the values of $U_i$ converge.\n",
+    "We terminate the algorithm if the $delta$ value is less than a threshold value determined by the hyperparameter _epsilon_.\n",
     "\n",
     "$$\\delta \\lt \\epsilon \\frac{(1 - \\gamma)}{\\gamma}$$\n",
     "\n",
-    "To summarize, the Bellman update is a _contraction_ by a factor of `gamma` on the space of utility vectors. \n",
-    "Hence, from the properties of contractions in general, it follows that `value_iteration` always converges to a unique solution of the Bellman equations whenever gamma is less than 1.\n",
+    "To summarize, the Bellman update is a _contraction_ by a factor of $gamma$ on the space of utility vectors. \n",
```

Comment:
I think this should not be changed as the backticks were used to denote that `gamma` is a variable used in the code.

Reply:
Makes sense. Changing it.

Reply:
Hmm, just went through this again and it seems to be a more mathematical/conceptual statement than an implementational one. In the sense, the $\gamma$ here isn't explaining the code, or even the pseudocode.

I think sticking to the symbol might be the better option. Thoughts?

Reply:
You may be right. It's been some time since I wrote this and I don't remember what I was thinking. You can go ahead.